l10n_es_account_asset_variation/models/account_asset.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Ainara Galdona - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+import calendar
+from openerp import fields, models, api
+from dateutil.relativedelta import relativedelta
+from openerp.addons import decimal_precision as dp
+
+
+class AccountAssetDepreciationLine(models.Model):
+
+    _inherit = 'account.asset.depreciation.line'
+
+    method_percentage = fields.Float(string='Method Percentage')
+
+    @api.multi
+    def write(self, values):
+        wiz = self.env.context.get('wiz', False)
+        if not wiz:
+            return super(AccountAssetDepreciationLine, self).write(values)
+        for record in self:
```

Comment:
Yo lo hubiera puesto así: 
if 'depreciation_date' in values:
   for record in self.filtered(lambda x: x.asset_id.method_time == 'percentage'):